### ISSUE TITLE
Improve resolution and value of collected stats...

### ISSUE THREAD

**Issue by sean-** [hashicorp/nomad#2107]
A few suggestions:

1) There should probably be three metric intervals: high, medium, and low frequency metrics.  High frequency metrics can be collected as frequently as once every 100ms (e.g. CPU, Go memory usage, etc).  Medium metrics once every 10s (I'm not sure what would need to be polled that infrequently), and low frequency metrics every ~60s or 300s (e.g. disk usage, etc).  It's been my experience that decoupling CPU from disk stats is important when the frequency is sub-second because of the number of locks acquired by the VFS to report out stats (whereas CPU counters are much cheaper).

2) All high and medium-frequency metrics of Gauge type should be converted from `SetGauge()` to `AddSample()`.

3) The conditional guarding the per-allocation metrics should remain in place.  Maybe a new nob, "collect high-frequency stats" that, when enabled, uses the high-frequency timeout loop, or when disabled, uses the medium-frequency timeout.

https://github.com/hashicorp/nomad/blob/8b6fcd34830a661951920070aaccc66d3377c4b0/client/client.go#L2092-L2116

**Comment by jippi** [comment]
Are you talking StatsD or nomads internal client and allocation stats?

**Comment by sean-** [comment]
No, this is not about statsd, this is internal to the client and allocation stats.  statsd would pick up any benefit from this change via its `armon/go-metrics` sink.
